instagram/src/components/CommentSection/CommentSection.js

```diff
@@ -0,0 +1,81 @@
+import React, {Component} from 'react';
+import Comment from '../Comment/Comment.js'
+import './CommentSection.css'
+import moment from 'moment';
+
+//Receives an individual post's comments array from PostContainer
+class CommentSection extends Component{
+    constructor(){
+        super();
+        this.state={
+            comments: [],
+            user: '',
+            addComment: '',
+            newComment: ''
+        }
+    }
+
+    componentDidMount(){
+        this.setState({
+            comments: this.props.comments,
+            user: this.props.user,
+            addComment: this.props.addComment
+
+        })
+    }
+
+    handleChange = (event) =>{
+        this.setState({
+            newComment : event.target.value
+        })
+    }
+
+
+    handleSubmit = (event) =>{
+        event.preventDefault();
+        // const commentInputBox = document.querySelector('.comment-input');
+        // let commentText = commentInputBox.value;
+        const newCommentObj = (this.state.newComment)
+        ? {
+            text: this.state.newComment,
+            username: 'critter73'
+        }
+        : null;
+
+        if(newCommentObj){
+            // debugger;
+            this.state.addComment(newCommentObj, this.state.user);
+            // commentInputBox.value = '';
+            this.state.newComment = '';
+        }
+    }
+
+    render(){
+        // console.log(moment.format());
+
+        const commentList = (!this.state.comments.length === 0)
+        ? null
+        : this.state.comments.map(comment =>{
+            return (
+                <div key={this.state.comments.username + Math.random().toString()}>
```

Comment:
Good idea here Christopher on generating unique keys. [Check out the React docs on lists and keys for other ideas.](https://reactjs.org/docs/lists-and-keys.html)